studio/components/interfaces/Auth/EmailTemplates.tsx

```diff
@@ -0,0 +1,266 @@
+import { Form, Input, Tabs } from '@supabase/ui'
+import CodeEditor from 'components/ui/CodeEditor'
+import InformationBox from 'components/ui/InformationBox'
+import { useStore } from 'hooks'
+import { observer } from 'mobx-react-lite'
+import { useEffect, useState } from 'react'
+import ReactMarkdown from 'react-markdown'
+import { TEMPLATES_SCHEMAS } from 'stores/authConfig/schema'
+import {
+  FormActions,
+  FormHeader,
+  FormPanel,
+  FormSection,
+  FormSectionContent,
+  FormSectionLabel,
+} from '../../ui/Forms'
+
+const EmailTemplates = observer(() => {
+  const { authConfig, ui } = useStore()
+
+  const isLoaded = authConfig.isLoaded
+
+  return (
+    <div>
+      <FormHeader
+        title="Email Templates"
+        description={`You can use your own SMTP server instead of the built-in email service.`}
+      />
+      <FormPanel>
+        <Tabs
+          defaultActiveId={TEMPLATES_SCHEMAS[0].title.trim().replace(/\s+/g, '-')}
+          type="underlined"
+          listClassNames="px-8 pt-4"
+          size="small"
+          scrollable
+        >
+          {TEMPLATES_SCHEMAS.map((template, i) => {
+            const [bodyValue, setBodyValue] = useState('')
+
+            const INITIAL_VALUES: { [x: string]: string } = {}
+
+            // console.log('template.properties', template.properties)
+
+            /**
+             * construct values for INITIAL_VALUES
+             *
+             * return empty string `""` rather than `null`
+             * as it breaks form. null is not a valid value.
+             */
+            Object.keys(template.properties).forEach((key) => {
```

Comment:
same - wrap in a useEffect? including the `delete` below

Reply:
im not sure it needs it. will take a look though